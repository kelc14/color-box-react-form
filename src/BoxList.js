import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import { v4 as uuid } from "uuid";

const BoxList = () => {
  const [boxes, setBoxes] = useState([]);

  /** Add the new box to state. */
  const addBox = (box) => {
    let newBox = { ...box, id: uuid() };
    setBoxes((boxes) => [...boxes, newBox]);
  };

  /** Remove the selected box from state. */
  const removeBox = (boxId) => {
    let noBox = boxes.filter((el) => el.id !== boxId);
    setBoxes(() => [...noBox]);
  };

  /** Render:
   *
   */
  return (
    <div className="BoxList">
      <NewBoxForm addBox={addBox} />
      {boxes.map((box) => (
        <Box
          key={box.id}
          id={box.id}
          backgroundColor={box.backgroundColor}
          width={box.width}
          height={box.height}
          removeBox={removeBox}
        />
      ))}
    </div>
  );
};

export default BoxList;
