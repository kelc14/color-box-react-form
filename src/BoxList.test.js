import { render, fireEvent, getByRole } from "@testing-library/react";
import BoxList from "./BoxList";

// smoke test
it("renders without crashing", function () {
  render(<BoxList />);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

// functionality tests
it("can add a new box", function () {
  const { getByLabelText, queryByText } = render(<BoxList />);

  // no boxes yet  (so no remove buttons)
  expect(queryByText("X")).not.toBeInTheDocument();

  const colorInput = getByLabelText("Color");
  const widthInput = getByLabelText("Width");
  const heightInput = getByLabelText("Height");
  const addBtn = queryByText("ADD");

  // fill out the form:
  fireEvent.change(colorInput, { target: { value: "red" } });
  fireEvent.change(widthInput, { target: { value: 100 } });
  fireEvent.change(heightInput, { target: { value: 200 } });
  fireEvent.click(addBtn);

  // item exists (remove button)
  expect(queryByText("X")).toBeInTheDocument();
});

// it("works when you click on the right arrow", function () {
//   const { container } = render(
//     <Carousel photos={TEST_IMAGES} title="images for testing" />
//   );
//   // expect the first image to show, but not the second
//   expect(
//     container.querySelector('img[alt="testing image 1"]')
//   ).toBeInTheDocument();
//   expect(
//     container.querySelector('img[alt="testing image 2"]')
//   ).not.toBeInTheDocument();

//   // move forward in the carousel
//   const rightArrow = container.querySelector(".bi-arrow-right-circle");
//   fireEvent.click(rightArrow);

//   // expect the second image to show, but not the first
//   expect(
//     container.querySelector('img[alt="testing image 1"]')
//   ).not.toBeInTheDocument();
//   expect(
//     container.querySelector('img[alt="testing image 2"]')
//   ).toBeInTheDocument();
// });

// it("works when you click on the left arrow", function () {
//   const { container } = render(
//     <Carousel photos={TEST_IMAGES} title="images for testing" />
//   );
//   // expect the first image to show, but not the second
//   expect(
//     container.querySelector('img[alt="testing image 1"]')
//   ).toBeInTheDocument();
//   expect(
//     container.querySelector('img[alt="testing image 2"]')
//   ).not.toBeInTheDocument();

//   // move forward in the carousel
//   const leftArrow = container.querySelector(".bi-arrow-left-circle");
//   fireEvent.click(leftArrow);
//   console.log(TEST_IMAGES);

//   // expect the second image to show, but not the first
//   expect(
//     container.querySelector('img[alt="testing image 1"]')
//   ).not.toBeInTheDocument();
//   expect(
//     container.querySelector('img[alt="testing image 3"]')
//   ).toBeInTheDocument();
// });
