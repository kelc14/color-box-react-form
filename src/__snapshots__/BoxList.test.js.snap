// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <div
      class="NewBoxForm"
    >
      <form
        class="NewBoxForm-form"
      >
        <label
          for="backgroundColor"
        >
          Color
        </label>
        <input
          id="backgroundColor"
          name="backgroundColor"
          type="text"
          value=""
        />
        <label
          for="width"
        >
          Width
        </label>
        <input
          id="width"
          name="width"
          type="number"
          value="0"
        />
        <label
          for="height"
        >
          Height
        </label>
        <input
          id="height"
          name="height"
          type="number"
          value="0"
        />
        <button>
          ADD
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
